dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/dxf2/sync/TrackerSynchronizationTest.java

```diff
@@ -110,16 +126,53 @@ private void prepareSyncParams()
     {
         queryParams = new TrackedEntityInstanceQueryParams();
         queryParams.setIncludeDeleted( true );
-        queryParams.setSynchronizationQuery( true );
         params = new TrackedEntityInstanceParams( false, TrackedEntityInstanceEnrollmentParams.FALSE, false, false,
             true, true );
     }
 
     @Test
-    void testSkipSyncFunctionality()
+    void shouldReturnAllTeisWhenNotSyncQuery()
+    {
+        queryParams.setSynchronizationQuery( false );
+        queryParams.setSkipChangedBefore( null );
+
+        List<org.hisp.dhis.dxf2.events.trackedentity.TrackedEntityInstance> fetchedTeis = subject
+            .getTrackedEntityInstances( queryParams, params, true, true );
+
+        assertContainsOnly( fetchedTeis.stream().map( t -> t.getTrackedEntityInstance() ).toList(),
+            List.of( TEI_UID, TEI_A_UID ) );
+        assertEquals( 1, getTeiByUid( fetchedTeis, TEI_UID ).getAttributes().size() );
+    }
+
+    @Test
+    void shouldNotSynchronizeTeiUpdatedBeforeLastSync()
     {
+        queryParams.setSynchronizationQuery( true );
+        queryParams.setSkipChangedBefore( null );
+
         List<org.hisp.dhis.dxf2.events.trackedentity.TrackedEntityInstance> fetchedTeis = subject
             .getTrackedEntityInstances( queryParams, params, true, true );
-        assertEquals( 1, fetchedTeis.get( 0 ).getAttributes().size() );
+
+        assertContainsOnly( fetchedTeis.stream().map( t -> t.getTrackedEntityInstance() ).toList(),
+            List.of( TEI_UID ) );
+        assertEquals( 1, getTeiByUid( fetchedTeis, TEI_UID ).getAttributes().size() );
+    }
+
+    @Test
+    void shouldNotSynchronizeTeiUpdatedBeforeSkipChangedBeforeDate()
+    {
+        queryParams.setSynchronizationQuery( true );
+        queryParams.setSkipChangedBefore( TOMORROW );
```

Comment:
Maybe the test setup could make that clearer why `TEI_UID` is not part of the result. It might also be that using `TOMORROW` is throwing me off 😁